scim.sgnl.yaml

```diff
@@ -0,0 +1,457 @@
+    parent: Users  # External ID of the parent entity
+    displayName: SCIMUserEntitlement
+    externalId: entitlements
+    description: Entity representing SCIM user entitlements
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: value
+        externalId: value
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: display
+        externalId: display
+        type: String  
+      - name: type
+        externalId: type
+        type: String
+      - name: primary
+        externalId: primary
+        type: Bool
+  
+  Roles:
+    parent: Users  # External ID of the parent entity
+    displayName: SCIMUserRole
+    externalId: roles
+    description: Entity representing SCIM user roles
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: value
+        externalId: value
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: display
+        externalId: display
+        type: String  
+      - name: type
+        externalId: type
+        type: String
+      - name: primary
+        externalId: primary
+        type: Bool
+
+  Photo:
+    parent: Users  # External ID of the parent entity
+    displayName: SCIMUserPhoto
+    externalId: photos
+    description: Entity representing SCIM user photos
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: value
+        externalId: value
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: display
+        externalId: display
+        type: String  
+      - name: type
+        externalId: type
+        type: String
+      - name: primary
+        externalId: primary
+        type: Bool
+  
+  Certificates:
+    parent: Users  # External ID of the parent entity
+    displayName: SCIMUserCertificate
+    externalId: certificates
+    description: Entity representing SCIM user certificates
+    syncFrequency: HOURLY
+    syncMinInterval: 1
+    apiCallFrequency: SECONDLY
+    apiCallMinInterval: 1
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: value
+        externalId: value
+        type: String
+        indexed: true
+        uniqueId: true
+      - name: display
+        externalId: display
+        type: String  
+      - name: type
+        externalId: type
+        type: String
+      - name: primary
+        externalId: primary
+        type: Bool
+
+relationships:
+  # Manager relationship
+  UserManager:
+    name: Manager
+    fromAttribute: Users.$..["urn:ietf:params:scim:schemas:extension:enterprise:2.0:User"].manager.value
+    toAttribute: Users.id
```

Comment:
```suggestion
    fromAttribute: User.managerId
    toAttribute: User.id
```

Reply:
This would result in an error when uploading to the UI as we expect `(entity.externalId).(attribute.externalId)`. Tested this live with Ivan.

cc: @ivanfong

Reply:
Wow, that sounds like a bug. Why are we not using the entity's logical name within the template?

Reply:
It's less confusion for both entity and attribute to use `externalId`. We could use the logical name for entity, but attributes only have name and externalId, so we just standardize and use externalId for both, since both have it. Also `id` being inherently unique is a plus.

I did not foresee having long externalIds such as this, and have created a story to tackle this problem: [sc-17038](https://app.shortcut.com/sgnl/story/17038/update-sor-template-format-to-support-weird-externalids#activity-17039)

@shash-sgnl lmk if i'm misremembering things

Reply:
How does this work when an entity has a JSONPath as its external ID?
How does the parsing work here?

Reply:
Yes; consistency was exactly the reason why externalId is being used for both. It was very error prone when sales/solutions tried to build templates. Thanks for bringing this up  @ivanfong  & @r-rakshith

Reply:
> Yeah, how does this work when an entity has a JSONPath as its external ID? How does the parsing work here?

Currently we just split it using the first `.` as delimiter. Then we match the first part to entity's externalId and second part to attribute's externalId.

Reply:
I'm fine with merging this as is since it's not a blocker. Since his pattern is being used in other templates as well, we can tackle it in sc-17038.